src/altinn-app-frontend/src/components/GenericComponent.tsx

```diff
@@ -154,6 +155,20 @@ export function GenericComponent<Type extends ComponentExceptGroup>(
     );
   }, [componentValidations]);
 
+  React.useLayoutEffect(() => {
+    if (!hidden && shouldFocus && gridRef.current) {
+      gridRef.current.scrollIntoView();
+
+      const maybeInput = gridRef.current.querySelector('input,textarea') as
```

Comment:
We probably should cater for other focusable items as well? Selects, radio, checkbox, etc...

Reply:
Yup - I never tested any other items. I'd prefer if there were ways to force these components themselves to implement focusing. For example, while the `DatePicker` contains an `input`, it might be best to put focus on the button to open the calendar instead. For `select`, I think it's important the dropdown doesn't actually open when setting focus, and radio/checkbox should default to focus the chosen value (if any). I'll look into adding more of the easy ones, at least.

Reply:
Done! `select` does not open the dropdown by default. Radio and checkbox already worked (they are `input`), but there is no visual indication these are focused.